packages/catalog-generator/src/test/java/io/kaoto/camelcatalog/generator/CamelYamlDslSchemaProcessorTest.java

```diff
@@ -60,7 +62,41 @@ public void testExtractSingleOneOfFromAnyOf() throws Exception {
 
         assertTrue(errorHandlerSchema.has("oneOf"));
         assertTrue(errorHandlerSchema.get("oneOf").isArray());
-        assertEquals(7, errorHandlerSchema.get("oneOf").size());
+        assertEquals(6, errorHandlerSchema.get("oneOf").size());
+    }
+
+    @Test
+    public void testRemoveEmptyProperties() throws Exception {
+        var tokenizerRootSchema = (ObjectNode) jsonMapper.readTree(
+                getClass().getClassLoader().getResourceAsStream("camel-4.9.0-tokenizer-schema.json"));
+        processor = new CamelYamlDslSchemaProcessor(jsonMapper, tokenizerRootSchema);
+
+        var subSchemaMap = processor.processSubSchema();
+        var tokenizerSchema = jsonMapper.readTree(subSchemaMap.get("tokenizer"));
+        var properties = tokenizerSchema.withObject("/properties");
+
+        assertFalse(properties.has("langChain4jCharacterTokenizer"));
+        assertFalse(properties.has("langChain4jLineTokenizer"));
+        assertFalse(properties.has("langChain4jParagraphTokenizer"));
+        assertFalse(properties.has("langChain4jSentenceTokenizer"));
+        assertFalse(properties.has("langChain4jWordTokenizer"));
```

Comment:
to help understand error failure, it is convenien tto provide an error message on the reason

Reply:
Added